### PR TITLE
✨(metabase) add `maps` data and notebook

### PR DESCRIPTION
**Pull request by loco-philippe**
Add maps for metropole and regions (level: department, epci, city) and creation notebook.

### CODE REVIEW

File: src/notebook/clean-static-dataset.md
Comment (jmaupetit) [inline src/notebook/clean-static-dataset.md:1]: This notebook has moved to `src/notebook/misc`! It should be remove from this PR :wink:

File: src/notebook/misc/maps.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    language: python
+    name: python3
+---
+
+# QualiCharge maps
+
+
+## Introduction
+
+L'utilisation de cartes choroplèthe pour la restitution d'indicateurs QualiCharge nécessite le chargement préalable des contours des zones affichées regroupées par cartes.
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:20]:
```suggestion
L'utilisation de cartes choroplèthes pour la restitution d'indicateurs QualiCharge nécessite le chargement préalable des contours des zones affichées regroupées par cartes.
```

File: src/notebook/misc/maps.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+Pour des raisons de facilité de consultation, les cartes utilisées sont séparées par zone géographique:
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:22]:
```suggestion
Pour des raisons de facilité de consultation, les cartes utilisées sont séparées par zone géographique :
```

File: src/notebook/misc/maps.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+- Territoire européen (régions européennes : 11, 24, 27, 28, 32, 44, 52, 53, 75, 76, 84, 93, 94)
+- Territoire américain (régions Guadeloupe, Martinique, Guyane : 1, 2, 3)
+- Territoire africain (régions La Réunion, Mayotte : 4, 6)
+
+Quatre niveaux de découpage sont proposés:
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:28]:
```suggestion
Quatre niveaux de découpage sont proposés :
```

File: src/notebook/misc/maps.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+- régions
+- départements
+- communes
+- agglomérations
+
+Le quatrième niveau est dérivé du découpage communal en intégrant les regroupements de communes définis dans les établissement public de coopération intercommunale (EPCI) à ficalité propre (métropoles, communautés urbaines, communautés d'agglomération, communautés de communes, Aix-Marseille et Grand Paris).
+
+Les régions sont découpées en départements. Les régions ultramarines (hors territoire européen) ne comportent qu'un seul département (code '97x' avec 'x' code région).
+Les départements sont découpés en communes.
+Les agglomérations sont des regroupements de communes
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:39]:
```suggestion
Les agglomérations sont des regroupements de communes.
```

File: src/notebook/misc/maps.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+
+
+## Map utilisées
+
+Les cartes utilisées pour restituer les indicateurs sont alors les suivantes (la métropole correspond au territoire européen):
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:45]:
```suggestion
Les cartes utilisées pour restituer les indicateurs sont alors les suivantes (la métropole correspond au territoire européen) :
```

File: src/notebook/misc/maps.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+  - metropole_reg.geojson
+  - metropole_dep.geojson
+  - region_xx_dep.geojson
+  - region_xx_epci.geojson
+  - region_xx_com.geojson
+
+  avec xx : code région
+
+Chaque contour (type : Polygon ou MultiPolygon) est associé à deux propriétés:
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:70]:
```suggestion
Chaque contour (type : Polygon ou MultiPolygon) est associé à deux propriétés:

```

File: src/notebook/misc/maps.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+for query, ext in zip(queries_reg, extension):
+    with engine.connect() as conn:
+        df = pd.read_sql_query(query, conn)
+    for reg in regions:
+        file_dep = 'region_' + reg + ext +'.geojson'
+        map_geo = {'type': 'FeatureCollection', 
+                   'features': [{'type': 'Feature', 
+                                 'properties': {'code': row[1], 'nom': row[2]}, 
+                                 'geometry': {'type': ('MultiPolygon' if str(row[3])[:4] == '[[[[' else 'Polygon'), 'coordinates': row[3]}} 
+                                for row in list(df.itertuples()) if row[4] == reg]}
```
Comment (jmaupetit) [inline src/notebook/misc/maps.md:152]: Maybe the following would more readable:

```suggestion
                                for row in df[df["code"] == reg].itertuples()}
```

**Review by jmaupetit — COMMENTED** [review]
Thanks! :pray:

**Comment by loco-philippe** [comment]
To take 'mixed EPCI' into account, `DISTINCT` is removed from requests and replaced by `drop_duplicates` for each region.

File: src/notebook/clean-static-dataset.md
Comment (loco-philippe) [reply src/notebook/clean-static-dataset.md:1]: removed !

**Pull request merged**